packages/telemetry/src/tooling-telemetry/data-processor.ts

```diff
@@ -262,29 +263,51 @@ function getInternalVsExternal(): InternalFeature {
 }
 
 /**
- * Read the manifest.json for the app and locate the tools id.
+ * Retrieves the source template configuration from either the standard manifest.json or
+ * the ui5.yaml based on the project type.
  *
- * @param appPath appPath Root folder path of Fiori app
- * @returns sourceTemplate section of data from manifest.json
+ * @param {string} appPath - The file system path to the application directory.
+ * @returns {Promise<SourceTemplate>} A promise that resolves to the source template configuration object.
  */
-async function getManifestSourceTemplate(appPath: string): Promise<SourceTemplate> {
-    let sourceTemplate: SourceTemplate = {};
+async function getSourceTemplate(appPath: string): Promise<SourceTemplate> {
+    const paths = {
+        manifest: path.join(appPath, 'webapp', 'manifest.json'),
+        appdescr: path.join(appPath, 'webapp', 'manifest.appdescr_variant'),
+        ui5Yaml: path.join(appPath, 'ui5.yaml')
+    };
+
     try {
-        const manifestPath = path.join(appPath, 'webapp', 'manifest.json');
-        if (fs.existsSync(manifestPath)) {
-            const manifestStr = await fs.promises.readFile(manifestPath, 'utf-8');
+        if (fs.existsSync(paths.manifest)) {
+            const manifestStr = await fs.promises.readFile(paths.manifest, 'utf-8');
             const manifest = JSON.parse(manifestStr);
-            sourceTemplate = manifest['sap.app']?.sourceTemplate;
+            return populateSourceTemplate(manifest['sap.app']?.sourceTemplate ?? {});
+        }
+
+        if (fs.existsSync(paths.appdescr)) {
```

Comment:
I think that here will be good also to include `&& fs.existsSync(paths.ui5Yaml)` in the condition of the if statement